packages/context/src/__tests__/acceptance/binding-config.acceptance.ts

```diff
@@ -0,0 +1,192 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {config, configBindingKeyFor, Context, ContextView} from '../..';
+
+describe('Context bindings - injecting configuration for bound artifacts', () => {
+  let ctx: Context;
+
+  beforeEach(givenContext);
+
+  it('binds configuration independent of binding', async () => {
+    // Bind configuration
+    ctx.configure('servers.rest.server1').to({port: 3000});
+
+    // Bind RestServer
+    ctx.bind('servers.rest.server1').toClass(RestServer);
+
+    // Resolve an instance of RestServer
+    // Expect server1.config to be `{port: 3000}
+    const server1 = await ctx.get<RestServer>('servers.rest.server1');
+
+    expect(server1.configObj).to.eql({port: 3000});
+  });
+
+  it('configures an artifact with a dynamic source', async () => {
+    // Bind configuration
+    ctx
+      .configure('servers.rest.server1')
+      .toDynamicValue(() => Promise.resolve({port: 3000}));
+
+    // Bind RestServer
+    ctx.bind('servers.rest.server1').toClass(RestServer);
+
+    // Resolve an instance of RestServer
+    // Expect server1.config to be `{port: 3000}
+    const server1 = await ctx.get<RestServer>('servers.rest.server1');
+    expect(server1.configObj).to.eql({port: 3000});
+  });
+
+  it('configures an artifact with alias', async () => {
+    // Configure rest server 1 to reference `rest` property of the application
+    // configuration
+    ctx
+      .configure('servers.rest.server1')
+      .toAlias(configBindingKeyFor('application', 'rest'));
+
+    // Configure the application
+    ctx.configure('application').to({rest: {port: 3000}});
+
+    // Bind RestServer
+    ctx.bind('servers.rest.server1').toClass(RestServer);
+
+    // Resolve an instance of RestServer
+    // Expect server1.config to be `{port: 3000}
+    const server1 = await ctx.get<RestServer>('servers.rest.server1');
+    expect(server1.configObj).to.eql({port: 3000});
+  });
+
+  it('allows configPath for injection', async () => {
+    class RestServerWithPort {
+      constructor(@config('port') public port: number) {}
+    }
+
+    // Bind configuration
+    ctx
+      .configure('servers.rest.server1')
+      .toDynamicValue(() => Promise.resolve({port: 3000}));
+
+    // Bind RestServer
+    ctx.bind('servers.rest.server1').toClass(RestServerWithPort);
+
+    // Resolve an instance of RestServer
+    // Expect server1.config to be `{port: 3000}
+    const server1 = await ctx.get<RestServerWithPort>('servers.rest.server1');
+    expect(server1.port).to.eql(3000);
+  });
+
+  const LOGGER_KEY = 'loggers.Logger';
+  it('injects a getter function to access config', async () => {
+    class Logger {
+      constructor(
+        @config.getter()
+        public configGetter: () => Promise<LoggerConfig | undefined>,
```

Comment:
Would the following work?

```suggestion
        public configGetter: Getter<LoggerConfig | undefined>, 
```

Also why are we expecting to receive `undefined` too? Is `@config.getter` enabling `{optional: true}` under the hood? If that's the case, then please 
- add a code comment here to explain this fact to readers
- add a new test (or enhance this test) to test what happens when `configGetter` is called but no configuration was bound